src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.VersionInfo.cs

```diff
@@ -47,8 +48,48 @@ internal enum CurlFeatures : int
         [DllImport(Libraries.HttpNative, EntryPoint = "HttpNative_GetSslVersionDescription")]
         internal static extern string GetSslVersionDescription();
 
-        internal const string OpenSsl10Description = "openssl/1.0";
+        internal const string OpenSslDescriptionPrefix = "OpenSSL/";
         internal const string SecureTransportDescription = "SecureTransport";
         internal const string LibreSslDescription = "LibreSSL";
+
+#if !SYSNETHTTP_NO_OPENSSL
+        private static readonly Lazy<string> s_requiredOpenSslDescription =
+            new Lazy<string>(() => DetermineRequiredOpenSslDescription());
+
+        private static readonly Lazy<bool> s_hasMatchingOpenSsl =
+            new Lazy<bool>(() => RequiredOpenSslDescription == GetSslVersionDescription());
+
+        internal static string RequiredOpenSslDescription => s_requiredOpenSslDescription.Value;
+        internal static bool HasMatchingOpenSslVersion => s_hasMatchingOpenSsl.Value;
+
+        private static string DetermineRequiredOpenSslDescription()
+        {
+            string version = Interop.OpenSsl.OpenSslVersionDescription();
+
+            // OpenSSL version description looks like this:
+            //
+            // OpenSSL 1.1.1 FIPS  11 Sep 2018
+            //
+            // libcurl's OpenSSL vtls backend ignores status in the version string.
+            // Major, minor, and fix are encoded (by libcurl) as unpadded hex
+            // (0 => "0", 15 => "f", 16 => "10").
+            //
+            // Patch is encoded as in the way OpenSSL would do it.
+
+            string prefix = "OpenSSL ";
+            if (version.StartsWith(prefix))
+            {
+                version = version.AsSpan(prefix.Length).ToString().Trim();
+            }
+            int end = version.IndexOf(" ");
+            if (end != -1)
+            {
+                version = version.AsSpan(0, end).ToString();
+            }
+            version = version.Trim();
```

Comment:
It seems like all of these version-chopping lines could stay in ReadOnlySpan&lt;char&gt; until the returning concat.